doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,790 @@
+    * DPU Health
+    * Alarms, Syslog
+    * Console
+* RMA
+    * Inventory
+
+The purpose of this document is to provide a framework to share the state, health, alarms of the DPUs, manage the DPUs by providing support to monitor, gracefully shutdown, restart them and the associated peripherals such as thermal sensors, cooling devices, LEDs, etc.
+
+The picture below highlights the PMON vertical and its association with other logics within the SONiC architecture.
+
+<p align="center"><img src="./images/pmon-vertical.svg"></p>
+
+## 2.	Requirements and Assumptions
+### 2.1.    Onboarding
+* The SmartSwitch host PMON should be able to Power Cycle, Shutdown, Reset, and rest the PCIe link per DPU or the entire system. The DPU_MODULE will behave like the LINE_CARD_MODULE with respect to these functions. 
+    #### SmartSwitch Expected PowerUp sequence:
+    * When the smartswitch device is booted, the host will boot first and leave the DPUs in power-down state by default.
+    * The user will powerup/boot the DPUs as needed.
+
+    #### SmartSwitch Expected PowerDown sequence:
+    * When the smartswitch device is completely shutdown, it will try to gracefully shutdown all the DPUs first. In the event of failure to graceful shutdown will force shutdown.
+    * Then the switch will power down itself gracefully (NPU/Chassis)
+
+* The DPU must provide additional information such as reboot cause, timestamp, etc as explained in the scheme once it boots its OS to DPU_STATE table.
+* When the DPU reboots itself, should log the reboot cause and update the previous-reboot-cause field in the ChassisStateDB when it boots up again
+* The reboot-cause history should provide a holistic view of the reboot cause of the SmartSwitch host CPU, the reboot-cause of all the DPUs
+* The DPUs should be uniquely identified and the DPU upon boot may get this ID from the host and identify itself.
+* Implement the required API enhancements and new APIs for DPU management (see details in design section)
+* SmartSwitch should use the existing SONiC midplane-interface model in modular chassis design for communication between the DPU and the NPU
+* SmartSwitch should extend the SONiC modular chassis design and treat the dpu-cards just like line-cards in existing design
+* Reboot
+    * Only cold reboot of DPUs is required, warm boot support is not required.
+### 2.2. Monitoring and Thermal Management
+* Dpu State
+    * The DPUs should provide their state to the host as the boot progression happens by updating the dpu state data in the DPU_STATE table in the host ChassisStateDB (explained in DB schema)
+    * DPUs should be able to store the data using a redis call
+    * The DPU must provide the state information once it boots its OS to DPU_STATE table.
+    * The SmartSwitch host PMON should be able to monitor the liveliness of the DPUs and when they go down should be able to take appropriate actions  and should try to gracefully recover the DPU when requested by the PMON
```

Comment:
liveness: is this something new? if yes, please explain the flow

Reply:
This has been there from the beginning.  This is the DPU_SATE data which can be used by a controller or a HA or LB logic to make decisions.  For example, if a DPU updates it data plane state to be down, the HA logic can do the cut over and inform the controller, upon receiving the cutover message the controller can inform the host PMON to reboot the DPU so that it is available for future use.